### ISSUE TITLE
Create a Config Module - Digital Agency

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/zeever#10]
The initial plan was to create a module that would bring together several recipes and some addtional code to provide the functionality and content for this project. I had planned to call the theme "Zeever" and the module "Zeever Config". The idea was that these two items would be enough to create a working site template that someone could build off of.

However, I'm now thinking that there might be more than one theme that supports a given package of config and content, so I should probably not tie the config and content too closely with a theme. In fact, I should probably create a module called "Digital Agency" and then support the Digital Agency package with the Zeever theme. Leaving other themes to also support the package. 

I'm now thinking that I will refer to this module as a package, because it's purpose will be to bring together several modules and recipes to provide a set of functionality. The Zeever theme will then support this package and other themes may choose to as well.

**Comment by stpaultim** [comment]
I got a start on this today. 

This package (module) simply does the following at this point. 

1) Reqires the Services Recipe
2) Requires the FAQ Recipe
3) Requires the Testimonials Recipe

When all three recipes and this package are available in the project browser, one will be able to easily use this package to install all three recipes and their sample content at once.

**Comment by izmeez** [comment]
I wonder if there may be a better name than `Digital Agency` because the sections, the components, being considered are more generic: who we are, what we do, testimonials, ... faq ... contact us.

**Comment by izmeez** [comment]
I'm still learning Backdrop and what differences between `content` and `cards`. Conceptually, I gravitate to the idea of using `cards` for the components. Can cards use `paragraphs`?

**Comment by izmeez** [comment]
If content is added, it may help for the recipe to add users, even one user, to avoid user1 as content author.

**Comment by stpaultim** [comment]
@izmeez Thanks for the feedback and questions. It's helpful. 

> I wonder if there may be a better name than Digital Agency because the sections, the components, being considered are more generic: who we are, what we do, testimonials, ... faq ... contact us.

I'm certainly open to other names. While you are correct that this recipe has elements that could be used for many different things, it also comes with some elements to list "services" and a "portfolio." These specific elements were added for something like an artist or digital agency. 

My goal is specifically NOT try and create a theme that strives for maximum flexibility. Rather, I'm experimenting with the idea of an opinioned package or theme that very much has some specific use cases in mind. Of course, it will still be possible to modify the site or theme and use it for other use cases, but the goal is to maximize it's use case for at least one use case. 

I choose Zeever and it's audience, because I think it would be a useful use case, package, and theme to have for Backdrop CMS. Here is how Zeever is described by it's original creators for Wordpress. (https://wordpress.org/themes/zeever/)

> Zeever is a modern, responsive, and elegant dark style theme template for WordPress fullsite editing and fully compatible with Gutenverse plugin. Zeever’s created especially for Digital Agency, Creative Agency, Online Portfolio, Designer and Freelancer website for companies or individuals that need a beautiful landing page for their business. The templates includes both core version and Gutenverse plugin version, also has core and Gutenverse block patterns ready so you can start mix and match your template parts as you desire. The templates is built ready so you don't need to build it from scratch. We want to make your experience using WordPress fullsite editor more convenient.

Here is the full list of Content Types that I expect to be available in this package: https://github.com/stpaultim/zeever/wiki/Content-Types

**Comment by stpaultim** [comment]
> If content is added, it may help for the recipe to add users, even one user, to avoid user1 as content author.

I don't think it makes sense to create a user as part of the "package", because users must have an email address and password and we won't have either of these things to use for this initial user. 

We have partially addressed this issue with core, by creating the "Editor" role. It's quite easy for the site builder to add a user and give them this role. I think that is probably good enough. 

However, I do see the intent of this suggestion and will think about it a little more.

**Comment by stpaultim** [comment]
@izmeez 
> I'm still learning Backdrop and what differences between content and cards. Conceptually, I gravitate to the idea of using cards for the components. Can cards use paragraphs?

Cards are one type of content, like Pages for Posts. What makes "Cards" different is that they are intended to be use as pathless nodes. They are intended to demonstrate and help people userstand that you can create content without a path, meaning that unauthenticated users will not be able to access them on a page of their own. They are nodes that are intended to be used in "existing content blocks" or "views." See this page for a bit more context:
https://docs.backdropcms.org/documentation/creating-and-editing-content

We called this content type "cards", because after much discussion we could not come up with anything better. And, because "cards" is term that is often used for blocks of content similar to the demonstrated use of our new "Cards" content type. 

In theory, one can do anything with Cards that one does with Pages or Posts. There would be nothing wrong with using Paragraphs with the Cards content type. We may decide to use Paragraphs as part of this recipe/package.

**Comment by stpaultim** [comment]
@izmeez and others. 

I went ahead and released a "Digital Agency" package/recipe. I went with this name, because it does describe the primary use case for the package, although it could be used in many other ways. 

I think that there will be time to rethink the name, but I wanted to get this into the Backdrop contrib space and issue a release to make it much easier for testing and feedback. 

Use this issue queue to discuss and provide feedback. But, here is the code: https://github.com/backdrop-contrib/digital_agency
